app/src/ui/cherry-pick/choose-target-branch.tsx

```diff
@@ -0,0 +1,173 @@
+import * as React from 'react'
+import { Branch } from '../../models/branch'
+import { IMatches } from '../../lib/fuzzy-find'
+import {
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  OkCancelButtonGroup,
+} from '../dialog'
+import { BranchList, IBranchListItem, renderDefaultBranch } from '../branches'
+import { ClickSource } from '../lib/list'
+
+interface IChooseTargetBranchDialogProps {
+  /**
+   * See IBranchesState.defaultBranch
+   */
+  readonly defaultBranch: Branch | null
+
+  /**
+   * The currently checked out branch
+   */
+  readonly currentBranch: Branch
+
+  /**
+   * See IBranchesState.allBranches
+   */
+  readonly allBranches: ReadonlyArray<Branch>
+
+  /**
+   * See IBranchesState.recentBranches
+   */
+  readonly recentBranches: ReadonlyArray<Branch>
+
+  /**
+   * A function that's called when the user selects a branch and hits start
+   * cherry pick
+   */
+  readonly onCherryPick: (targetBranch: Branch) => void
+
+  /**
+   * A function that's called when the dialog is dismissed by the user in the
+   * ways described in the Dialog component's dismissable prop.
+   */
+  readonly onDismissed: () => void
+}
+
+interface IChooseTargetBranchDialogState {
+  /** The currently selected branch. */
+  readonly selectedBranch: Branch | null
+
+  /** The filter text to use in the branch selector */
+  readonly filterText: string
+}
+
+/** A component for initiating a rebase of the current branch. */
+export class ChooseTargetBranchDialog extends React.Component<
+  IChooseTargetBranchDialogProps,
+  IChooseTargetBranchDialogState
+> {
+  public constructor(props: IChooseTargetBranchDialogProps) {
+    super(props)
+
+    this.state = {
+      selectedBranch: null,
+      filterText: '',
+    }
+  }
+
+  private onFilterTextChanged = (filterText: string) => {
+    this.setState({ filterText })
+  }
+
+  private onSelectionChanged = (selectedBranch: Branch | null) => {
+    this.setState({ selectedBranch })
+  }
+
+  private renderBranch = (item: IBranchListItem, matches: IMatches) => {
+    return renderDefaultBranch(item, matches, this.props.currentBranch)
+  }
+
+  private onEnterPressed = (branch: Branch, source: ClickSource) => {
+    if (source.kind !== 'keyboard' || source.event.key !== 'Enter') {
+      return
+    }
+
+    source.event.preventDefault()
+
+    const { selectedBranch } = this.state
+
+    if (selectedBranch !== null && selectedBranch.name === branch.name) {
```

Comment:
This should also check if the selected branch is the current branch, right?
```suggestion
    if (
      selectedBranch !== null &&
      selectedBranch.name === branch.name &&
      this.canCherryPickOntoSelectedBranch()
    ) {
```

Reply:
No. Current branch is the branch that is currently checked out that we are cherry picking from. This method is when the user hits enter. It is for the flow such that a user clicks on a "target-branch" to cherry pick to and  then hits enter to start the cherry pick. The branch that is a parameter should be the selected branch; this is just a fail safe in case of a timing issue of user clicks branch A and then clicks branch B at the same time they hit enter. Enter branch would be B but state has not updated to A yet. (forcing them to hit enter again when the state has actually updated and thus the ui truly reflects the selected branch)

Reply:
Yes, but still you have to check the user didn't press enter in the current branch before calling `startCherryPick`, right?

I see you have added that check inside of `startCherryPick` which should be 👍 

Thank you!